clippy_lints/src/methods/mod.rs

```diff
@@ -2143,7 +2143,8 @@ fn lint_single_char_pattern<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, _expr: &'tcx h
         then {
             let mut applicability = Applicability::MachineApplicable;
             let snip = snippet_with_applicability(cx, arg.span, "..", &mut applicability);
-            let hint = format!("'{}'", &snip[1..snip.len() - 1]);
```

Comment:
I think we should use the automatic escaping of the debug implementation. So instead of using `snippet_with_applicability`, we can just use `r.as_str().chars().next.unwrap()` to obtain the only `char` in the string and print it with `to_string()`.

Reply:
Ah, good point. Let me fix it...

Reply:
I'm trying with `let hint = format!("{:?}", r.as_str().chars().next().unwrap());`, but this causes issue with hexadecimal values like `\x03`, which is printed as `\u{3}`.

I hope I didn't misunderstand your comment.

Reply:
is that a problem? Or is it just the change of output that seems problematic?

Reply:
I believe the change of output might be problematic. In my opinion, the hint should only tell the developer to change the string to a char, instead of also suggesting to change the notation.

But if that's okay with you, I can push the changes.

Reply:
hmm... that's true. And everything that was not a `'` was already correctly escaped in the `""`, so your original fix is actually perfectly correct.